File: server/model/response/login_log.go
package response

// LoginLogResponse 登录日志响应
type LoginLogResponse struct {
	ID        int64  `json:"id"`
	Username  string `json:"username"`
	IP        string `json:"ip"`
	Location  string `json:"location"`
	Browser   string `json:"browser"`
	OS        string `json:"os"`
	Status    int    `json:"status"`
	Msg       string `json:"msg"`
	CreatedAt string `json:"createdAt"`
}

File: web/src/api/loginLog.ts
import request from '@/utils/request'
import type { ApiResponse, PageResponse } from '@/types/api'

/** 登录日志项 */
export interface LoginLogItem {
  id: number
  username: string
  ip: string
  location: string
  browser: string
  os: string
  status: number
  msg: string
  createdAt: string
}

/** 登录日志查询参数 */
export interface LoginLogListParams {
  page: number
  pageSize: number
  username?: string
  ip?: string
  status?: number
  startTime?: string
  endTime?: string
}

/** 获取登录日志列表 */
export function getLoginLogList(params: LoginLogListParams) {
  return request.get<PageResponse<LoginLogItem>>('/monitor/login-log/list', { params })
}

/** 清空登录日志 */
export function clearLoginLog() {
  return request.delete<ApiResponse<null>>('/monitor/login-log/clear')
}
